src/main/java/com/aws/greengrass/builtin/services/pubsub/PubSubIPCEventStreamAgent.java

```diff
@@ -50,7 +48,7 @@
     private static final String GLOB_WILDCARD = "*";
     private static final ObjectMapper SERIALIZER = new ObjectMapper();
     @Getter(AccessLevel.PACKAGE)
-    private final Map<String, Set<Object>> listeners = new ConcurrentHashMap<>();
```

Comment:
This was a `ConcurrentHashMap`. Does this mean our new Trie operations must be thread-safe as well?